hypothesis-python/tests/cover/test_type_lookup.py

```diff
@@ -291,3 +293,66 @@ def test_generic_origin_concrete_builds():
         assert_all_examples(
             st.builds(using_generic), lambda example: isinstance(example, int)
         )
+
+
+class AbstractFoo(abc.ABC):
+    def __init__(self, x):
+        pass
+
+    @abc.abstractmethod
+    def qux(self):
+        pass
+
+
+class ConcreteFoo1(AbstractFoo):
+    # Can't resolve this one due to unannotated `x` param
+    def qux(self):
+        pass
+
+
+class ConcreteFoo2(AbstractFoo):
+    def __init__(self, x: int):
+        pass
+
+    def qux(self):
+        pass
+
+
+@given(st.from_type(AbstractFoo))
+def test_gen_abstract(foo):
+    # This requires that we correctly checked which of the subclasses
+    # could be resolved, rather than unconditionally using all of them.
+    assert isinstance(foo, ConcreteFoo2)
+
+
+class AbstractBar(abc.ABC):
+    def __init__(self, x):
+        pass
+
+    @abc.abstractmethod
+    def qux(self):
+        pass
+
+
+class ConcreteBar(AbstractBar):
+    def qux(self):
+        pass
+
+
+def test_abstract_resolver_fallback():
+    # We create our distinct strategies for abstract and concrete types
+    gen_abstractbar = _from_type(AbstractBar)
```

Comment:
Why do we use different builders here? The first one is `_from_type`, the second is `st.build`.

Reply:
`st.builds()` to get a strategy for `ConcreteBar`, which `from_type()` doesn't work on (because the `x` param is required but unannotated).  
`_from_type()` because we want to exercise exactly that logic, without the usual cache/retry logic that `st.from_type()` wraps it in.